contracts/contracts/Subcall.sol

```diff
@@ -0,0 +1,219 @@
+                                contextIdentifier,
+                                contextVersion,
+                                data
+                            )
+                        )
+                    )
+                )
+            );
+    }
+}
+
+library Subcall {
+    address internal constant SUBCALL =
+        0x0100000000000000000000000000000000000103;
+
+    error Subcall_Error();
+
+    /**
+     * Submit a native message to the Oasis runtime layer
+     *
+     * Messages which re-enter the EVM module are forbidden: evm.*
+     *
+     * @param method Native message type
+     * @param body CBOR encoded body
+     * @return status Result of call
+     * @return data CBOR encoded result
+     */
+    function subcall(string memory method, bytes memory body)
+        internal
+        returns (uint64 status, bytes memory data)
+    {
+        (bool success, bytes memory tmp) = SUBCALL.call(
+            abi.encode(method, body)
+        );
+
+        if (false == success) {
+            revert Subcall_Error();
+        }
+
+        (status, data) = abi.decode(tmp, (uint64, bytes));
+    }
+
+    function _subcallWithToAndAmount(
+        string memory method,
+        StakingPublicKey to,
+        uint128 value
+    ) internal returns (uint64 status, bytes memory data) {
+        (status, data) = subcall(
+            method,
+            abi.encodePacked(
+                hex"a262",
+                "to",
+                hex"55",
+                to,
+                hex"66",
+                "amount",
+                hex"8250",
+                value,
+                hex"40"
+            )
+        );
+    }
+
+    error ConsensusUndelegateError(uint64 status, string data);
+
+    /**
+     * Start the undelegation process of the given number of shares from
+     * consensus staking account to runtime account.
+     *
+     * @param from Public key which shares were delegated to
+     * @param shares Number of shares to withdraw back to us
+     */
+    function consensusUndelegate(StakingPublicKey from, uint128 shares)
+        internal
+    {
+        (uint64 status, bytes memory data) = subcall(
+            "consensus.Undelegate",
+            abi.encodePacked(
+                hex"a264",
+                "from",
+                hex"55",
+                from,
+                hex"66",
+                "shares",
+                hex"50",
+                shares
+            )
+        );
+
+        if (status != 0) {
+            revert ConsensusUndelegateError(status, string(data));
+        }
+    }
+
+    error ConsensusDelegateError(uint64 status, string data);
+
+    /**
+     * Delegate native token to consensus level
+     *
+     * @param to Staking account
+     * @param value native token amount (in wei)
+     */
+    function consensusDelegate(StakingPublicKey to, uint128 value) internal {
+        (uint64 status, bytes memory data) = _subcallWithToAndAmount(
+            "consensus.Delegate",
```

Comment:
Move to a constant above, so we have all method names in one place.

FYI go, typescript and rust bindings:
- https://github.com/oasisprotocol/oasis-sdk/tree/main/client-sdk/go/modules
- https://github.com/oasisprotocol/oasis-sdk/tree/main/client-sdk/ts-web/rt/src
- https://github.com/oasisprotocol/oasis-sdk/tree/main/runtime-sdk/src/modules